src/shims/sync.rs

```diff
@@ -318,7 +341,20 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
 
         let kind = this.read_scalar(kind_op)?.not_undef()?;
-        if kind == this.eval_libc("PTHREAD_MUTEX_NORMAL")?
+        if kind == this.eval_libc("PTHREAD_MUTEX_NORMAL")? {
+            // In `glibc` implementation, the numeric values of
+            // `PTHREAD_MUTEX_NORMAL` and `PTHREAD_MUTEX_DEFAULT` are equal, but
+            // they have different behaviour in some cases. Therefore, we add
```

Comment:
Well, passing them to `settype` still has the same behavior either way, of course, as they cannot be distinguished.

But *not setting anything* has a different behavior than setting `NORMAL` aka `DEFAULT`.